core/src/smartcontracts/isi/query.rs

```diff
@@ -33,33 +33,12 @@ impl ValidQueryRequest {
     }
 }
 
-/// Unsupported version error
```

Comment:
While I think we might need to bring this back when we have fully-functional versioning, removing it for the time being is fine.

Reply:
Then we can utilize `iroha_version::error::Error::UnsupportedVersion`.
The idea is that Version error is considered a part of Decode error

Reply:
Then we can utilize `iroha_version::error::Error::UnsupportedVersion`.
The idea is that Version error is considered a part of Decode error

Reply:
This is a bit of a problem in the code base in general. We have double coverage because one particular thing is not general-enough.

Reply:
I think keeping unused `QueryError::Version(UnsupportedVersionError)` is just misleading
because we already use `QueryError::Decode(iroha_version::error::Error::UnsupportedVersion(UnsupportedVersion))` for such cases